### ISSUE TITLE
The "#" character inside string or regexp literal is broken

### ISSUE THREAD

**Issue by aycabta** [jruby/jruby#5230]
### Environment

Provide at least:
* `jruby -v`: `jruby 9.2.0.0 (2.5.0) 2018-05-24 81156a8 OpenJDK 64-Bit Server VM 9.0.4+11 on 9.0.4+11 +jit [linux-x86_64]`
* `uname -a`: `Linux x270 4.15.15-1-ARCH #1 SMP PREEMPT Sat Mar 31 23:59:25 UTC 2018 x86_64 GNU/Linux`

### Expected Behavior

On CRuby 2.5.1:

```ruby
$ irb
irb(main):001:0> RUBY_VERSION
=> "2.5.1"
irb(main):002:0> require 'ripper'
=> true
irb(main):003:0> pp Ripper.lex('["\#", "\#{}", /\#/, /\#{}/, "#", "#{}", /#/, /#{}/]')
[[[1, 0], :on_lbracket, "[", EXPR_BEG|EXPR_LABEL],
 [[1, 1], :on_tstring_beg, "\"", EXPR_BEG|EXPR_LABEL],
 [[1, 2], :on_tstring_content, "\\#", EXPR_BEG|EXPR_LABEL],
 [[1, 4], :on_tstring_end, "\"", EXPR_END|EXPR_ENDARG],
 [[1, 5], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 6], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 7], :on_tstring_beg, "\"", EXPR_BEG|EXPR_LABEL],
 [[1, 8], :on_tstring_content, "\\\#{}", EXPR_BEG|EXPR_LABEL],
 [[1, 12], :on_tstring_end, "\"", EXPR_END|EXPR_ENDARG],
 [[1, 13], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 14], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 15], :on_regexp_beg, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 16], :on_tstring_content, "\\#", EXPR_BEG|EXPR_LABEL],
 [[1, 18], :on_regexp_end, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 19], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 20], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 21], :on_regexp_beg, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 22], :on_tstring_content, "\\\#{}", EXPR_BEG|EXPR_LABEL],
 [[1, 26], :on_regexp_end, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 27], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 28], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 29], :on_tstring_beg, "\"", EXPR_BEG|EXPR_LABEL],
 [[1, 30], :on_tstring_content, "#", EXPR_BEG|EXPR_LABEL],
 [[1, 31], :on_tstring_end, "\"", EXPR_END|EXPR_ENDARG],
 [[1, 32], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 33], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 34], :on_tstring_beg, "\"", EXPR_BEG|EXPR_LABEL],
 [[1, 35], :on_embexpr_beg, "\#{", EXPR_BEG|EXPR_LABEL],
 [[1, 37], :on_embexpr_end, "}", EXPR_END],
 [[1, 38], :on_tstring_end, "\"", EXPR_END|EXPR_ENDARG],
 [[1, 39], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 40], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 41], :on_regexp_beg, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 42], :on_tstring_content, "#", EXPR_BEG|EXPR_LABEL],
 [[1, 43], :on_regexp_end, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 44], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 45], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 46], :on_regexp_beg, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 47], :on_embexpr_beg, "\#{", EXPR_BEG|EXPR_LABEL],
 [[1, 49], :on_embexpr_end, "}", EXPR_END],
 [[1, 50], :on_regexp_end, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 51], :on_rbracket, "]", EXPR_END]]
```

The `#` inside string or regexp literal should be `:on_tstring_content`.

### Actual Behavior

On JRuby 9.2.0.0:

```ruby
$ irb
RUBY_VERirb(main):001:0> JRUBY_VERSION
=> "9.2.0.0"
irb(main):002:0> require 'ripper'
=> true
irb(main):003:0> pp Ripper.lex('["\#", "\#{}", /\#/, /\#{}/, "#", "#{}", /#/, /#{}/]')
[[[1, 0], :on_lbracket, "[", EXPR_BEG|EXPR_LABEL],
 [[1, 1], :on_tstring_beg, "\"", EXPR_BEG|EXPR_LABEL],
 [[1, 2], :on_tstring_content, "\\#", EXPR_BEG|EXPR_LABEL],
 [[1, 4], :on_tstring_end, "\"", EXPR_END|EXPR_ENDARG],
 [[1, 5], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 6], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 7], :on_tstring_beg, "\"", EXPR_BEG|EXPR_LABEL],
 [[1, 8], :on_tstring_content, "\\\#{}", EXPR_BEG|EXPR_LABEL],
 [[1, 12], :on_tstring_end, "\"", EXPR_END|EXPR_ENDARG],
 [[1, 13], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 14], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 15], :on_regexp_beg, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 16], :on_tstring_content, "\\#", EXPR_BEG|EXPR_LABEL],
 [[1, 18], :on_regexp_end, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 19], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 20], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 21], :on_regexp_beg, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 22], :on_tstring_content, "\\\#{}", EXPR_BEG|EXPR_LABEL],
 [[1, 26], :on_regexp_end, "/", EXPR_BEG|EXPR_LABEL],
 [[1, 27], :on_comma, ",", EXPR_BEG|EXPR_LABEL],
 [[1, 28], :on_sp, " ", EXPR_BEG|EXPR_LABEL],
 [[1, 29], :on_tstring_beg, "\"", EXPR_BEG|EXPR_LABEL],
 [[1, 30], :on_tstring_content, "#\", ", EXPR_BEG|EXPR_LABEL],
 [[1, 34], :on_tstring_end, "\"", EXPR_END|EXPR_ENDARG],
 [[1, 35], :on_comment, "\#{}\", /#/, /\#{}/]", EXPR_END|EXPR_ENDARG]]
```

The escaped `\#` inside string or regexp literal is correct, but `#` inside string or regexp literal is broken.

This behavior breaks RDoc.

**Comment by enebo** [comment]
This is fixed by #5210. Marking as dup.

**Comment by aycabta** [comment]
Thank you for your confirmation.
